Index: popt-1.10/Makefile.in
===================================================================
--- popt-1.10.orig/Makefile.in	2006-08-08 01:41:13.087464093 +0100
+++ popt-1.10/Makefile.in	2006-08-08 01:41:40.093763717 +0100
@@ -283,7 +283,7 @@
 
 TESTS = testit.sh
 include_HEADERS = popt.h
-usrlibdir = $(libdir)@MARK64@
+usrlibdir = $(libdir)
 usrlib_LTLIBRARIES = libpopt.la
 libpopt_la_SOURCES = popt.c findme.c poptparse.c poptconfig.c popthelp.c
 libpopt_la_LDFLAGS = @INTLLIBS@
