Fix autoconf 2.70 obsolete macros

--- a/configure.ac
+++ b/configure.ac
@@ -8,8 +8,9 @@
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([1.10 foreign -Wall])
 
-AC_PROG_CC_STDC
 AC_PROG_CC
+# AC_PROG_CC_STDC was merged in AC_PROG_CC in autoconf 2.70
+m4_version_prereq([2.70], [], [AC_PROG_CC_STDC])
 AC_USE_SYSTEM_EXTENSIONS
 AM_PROG_AR
 
@@ -32,7 +33,7 @@
         ;;
 esac
 AC_ARG_ENABLE([ld-version-script],
-              AC_HELP_STRING([--enable-ld-version-script],
+              AS_HELP_STRING([--enable-ld-version-script],
                              [enable/disable use of linker version script.
                               (default is system dependent)]),
               [have_ld_version_script=$enableval],
